tt_eager/tt_dnn/op_library/run_operation.hpp

```diff
@@ -401,4 +402,256 @@ std::vector<Device*> get_workers_for_op_output(const std::vector<Tensor>&& input
 
 } //namespace operation
 
+bool validate_worker_modes(const std::vector<Device*>& workers);
+
+class TensorMonad {
+
+    static void launch_async(
```

Comment:
Not clear why launch_async is required is the entire composite function is handed off the worker thread.

The composite is then run serially by the worker , without any races or hazards, since everything runs sequentially.

Reply:
If we have two ops.  Op A and Op B.   Op B depends on the output tensors of Op A before it can run.  Op A has to be completed before Op B can use those output tensors as input tensors and be created.   Unfortunately the prior op may not have all its output tensors on the same device as the next ops needs.  So in this case we allow the operation to fallback to the main thread instead of erroring out like launch_op does.